src/components/CheckIn.tsx

```diff
@@ -1,12 +1,21 @@
 import { TableCell, TableRow, Typography } from "@mui/material"
-import { PropsWithChildren } from "react"
-import { CheckIn } from "../models/checkin"
+import axios from "axios"
+import { PropsWithChildren, useEffect, useState } from "react"
+import { CheckIn, Event } from "../models/checkin"
+import { get_all_checkins_by_id } from "../services/CheckIn"
+
+export const CheckInRow = ({ checkin }: PropsWithChildren<{ checkin: CheckIn }>) => {   
+  
+  const get_all_events = async () => {
+      let all_events: Array<Event> = (await axios.get("http://localhost:3001/v1/event")).data;
+      const match_events_to_checkin = all_events.filter((event) => event.id == checkin.event)
```

Comment:
Okay, first of all, incredible job in thinking about this problem!

1. `filter` takes an array and returns a subset of that array that matches a given condition. We would prefer in this case to use `find` (functions the same way) because it only returns the *first* match.
You can think of it like `filter` takes `[A]` and returns `[B]` (where `B` is a subset of `A`), while `find` takes `[A]` and returns the first `C` that matches a given condition.